### PR TITLE
make bundle support configurable from a sync server

### PR DESCRIPTION
**Pull request by tburgin**
This pull is branched from https://github.com/google/santa/pull/168, ensure that 168 has been submitted before processing this request. I am mainly doing this now as to not forget.

*  santactl/sync: sync server enables/disables client bundle support

### CODE REVIEW

File: Source/common/SNTConfigurator.h [hunk 1]
```diff
@@ -159,22 +159,10 @@ extern NSString *const kDefaultConfigFilePath;
 @property(readonly, nonatomic) NSString *machineID;
 
 ///
-///  The number of seconds in-between full syncs while connected to FCM
+///  If YES, enables bundle detection for blocked events. This property is not stored on disk.
+///  It's value is set by a sync server that supports bundles. Defaults to NO.
```
Comment (tburgin) [inline Source/common/SNTConfigurator.h]: Done.

File: Source/common/SNTStoredEvent.h [hunk 2]
```diff
@@ -37,6 +37,12 @@
 @property NSString *filePath;
 
 ///
+///  Set to YES if the event is apart of a bundle. When an event is passed to SantaGUI this propery
```
Comment (tburgin) [inline Source/common/SNTStoredEvent.h]: Done.

File: Source/common/SNTStoredEvent.h [hunk 3]
```diff
@@ -37,6 +37,12 @@
...
+///  will be used as an idicator to to kick off bundle hashing as necessary. Default value is NO.
```
Comment (tburgin) [inline Source/common/SNTStoredEvent.h]: Done.

File: Source/common/SNTXPCControlInterface.h [hunk 4]
```diff
@@ -80,6 +80,8 @@
 - (void)setSyncCleanRequired:(BOOL)cleanReqd reply:(void (^)())reply;
 - (void)setWhitelistPathRegex:(NSString *)pattern reply:(void (^)())reply;
 - (void)setBlacklistPathRegex:(NSString *)pattern reply:(void (^)())reply;
+- (void)enableBundles:(void (^)(BOOL))reply;
+- (void)setEnableBundles:(BOOL)enableBundles reply:(void (^)())reply;
```
Comment (tburgin) [inline Source/common/SNTXPCControlInterface.h]: Agreed.

File: Source/santactl/Commands/SNTCommandStatus.m [hunk 5]
```diff
@@ -177,6 +187,7 @@ + (void)runWithArguments:(NSArray *)arguments daemonConnection:(SNTXPCConnection
       printf("  %-25s | %s\n", "Last Successful Rule Sync", [lastRuleSyncSuccessStr UTF8String]);
       printf("  %-25s | %s\n", "Push Notifications",
              (pushNotifications ? "Connected" : "Disconnected"));
+      printf("  %-25s | %s\n", "Enable Bundles", (enableBundles ? "Yes" : "No"));
```
Comment (tburgin) [inline Source/santactl/Commands/SNTCommandStatus.m]: Done.

File: Source/santad/SNTDaemonControlController.m [hunk 6]
```diff
@@ -180,8 +184,13 @@ - (void)setBlacklistPathRegex:(NSString *)pattern reply:(void (^)())reply {
...
 }
 
-- (void)watchdogInfo:(void (^)(uint64_t, uint64_t, double, double))reply {
-  reply(watchdogCPUEvents, watchdogRAMEvents, watchdogCPUPeak, watchdogRAMPeak);
+- (void)enableBundles:(void (^)(BOOL))reply {
+  reply([SNTConfigurator configurator].enableBundles);
+}
+
+- (void)setEnableBundles:(BOOL)enableBundles reply:(void (^)())reply {
+  [SNTConfigurator configurator].enableBundles = enableBundles;
```
Comment (tburgin) [inline Source/santad/SNTDaemonControlController.m]: Created a new issue for this:
https://github.com/google/santa/issues/174

File: Source/santad/SNTExecutionController.m [hunk 7]
```diff
@@ -191,22 +193,24 @@ - (void)validateBinaryWithMessage:(santa_message_t)message {
...
-                          @"\033[1mParent:    \033[0m %@ (%@)\n\n",
-            se.filePath, se.fileSHA256, se.parentName, se.ppid];
-        NSURL *detailURL = [SNTBlockMessage eventDetailURLForEvent:se];
-        if (detailURL) {
-          [msg appendFormat:@"More info:\n%@\n\n", detailURL.absoluteString];
-        }
-        [self printMessage:msg toTTYForPID:message.ppid];
-
-        [_notifierQueue addEvent:se customMessage:cd.customMsg];
+        dispatch_async(_eventQueue, ^{
```
Comment (tburgin) [inline Source/santad/SNTExecutionController.m]: To ensure the needsBundleHash property on the StoredEvent has been set before archiving the object and sending over to the GUI.

**Comment by tburgin** [comment]
PTAL

File: Source/santad/SNTExecutionController.m (hunk 7 above)
Comment (russellhancox) [inline Source/santad/SNTExecutionController.m]: As discussed, given that the property should always have been set before reaching this code, neither should need to be on the queue. Unless I'm missing something.

**Comment by tburgin** [comment]
PTAL

**Pull request merged**